blocks/form/constant.js

```diff
@@ -3,9 +3,60 @@ export const dragDropText = 'Drag and Drop To Upload';
 
 export const DEFAULT_THANK_YOU_MESSAGE = 'Thank you for your submission.';
 
+// Logging Configuration
+// Control logging via URL parameter: ?log=<level>
+// ?log=on → returns 'warn'
+// ?log=debug, info, error, off, warn → returns that level
+// Invalid values return 'warn' (fallback)
+// OR automatically enabled for AEM preview URLs (*.page) → returns 'warn'
+const VALID_LOG_LEVELS = ['error', 'debug', 'warn', 'info', 'off'];
+
+export const getLogLevelFromURL = (urlString = null) => {
+  // Semantic constants for log level defaults
+  const DEFAULT_LOG_LEVEL = 'off'; // Used when no logging is explicitly requested
+  const FALLBACK_LOG_LEVEL = 'warn'; // Used for invalid/empty values or AEM preview
+  
+  try {
+    // Extract URL object from either parameter or current context
+    let url;
+    if (urlString) {
+      // Explicit URL string provided (for workers - they need page URL passed from main thread)
+      url = new URL(urlString);
+    } else if (typeof window !== 'undefined' && window.location) {
+      // Main thread context - use page URL
+      url = new URL(window.location.href);
+    } else {
+      return DEFAULT_LOG_LEVEL; // No URL available
+    }
+
+    const { searchParams, hostname } = url;
+
+    // Priority 1: Check for explicit log parameter
+    const logParam = searchParams.get('log');
+    if (logParam !== null) {
+      // Special case: log=on means warn
+      if (logParam === 'on') {
```

Comment:
you do not need to check this. If logParam is not valid (in this case `on`) it will always return `warn`

Reply:
agree, done